test/vmtests/vmtests/utils/libvirt_utils.py

```diff
@@ -14,8 +18,99 @@ def __init__(self, name: str, memory_mib: int, core_count: int, os_disk_path: Pa
         self.os_disk_path: Path = os_disk_path
 
 
+def get_libvirt_machine_type(
+    libvirt_conn: libvirt.virConnect,
+) -> str:
+    domain_capabilities_str = libvirt_conn.getDomainCapabilities(None, None, None, None, 0)
+    domain_capabilities = ET.fromstring(domain_capabilities_str)
+    return domain_capabilities.find("./machine").text
+
+
+def get_libvirt_firmware_config(
+        libvirt_conn: libvirt.virConnect,
+        machine_type: str,
+        enable_secure_boot: bool,
+    ) -> Dict[str, Any]:
+        # Resolve the machine type to its full name.
+        domain_caps_str = libvirt_conn.getDomainCapabilities(
+            machine=machine_type, virttype="kvm"
```

Comment:
Just hardcode `machine="q35"`.

`pc-i440fx` is the old firmware that doesn't support things like secure-boot.